sematic/api/endpoints/tests/test_metrics.py

```diff
@@ -159,7 +160,16 @@ def test_get_metrics_endpoint(
 
     payload = response.json
 
-    assert payload["content"] == expected_series  # type: ignore
+    assert set(expected_series.keys()) == set(payload["content"].keys())  # type: ignore
+    assert all(
+        payload["content"][k] == expected_series[k]  # type: ignore
+        for k in expected_series.keys()
+        if k != "series"
+    )
+    check_approximate_equality(
```

Comment:
The logic was comparing strings (even before I changed the timestamps to strings). But sometimes the values were stringified floats and the string representation changed for some floating point values in sqlalchemy 2 (I *believe* because it started using the python `Decimal` type for fixed precision stuff and started adding trailing `0`s accordingly). Anyway, this compares normal strings by exact match and string-ified real numbers by making sure their percent diff is smaller than a small epsilon.